src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py

```diff
@@ -599,15 +599,23 @@ def _update_lock_parameters(parameters, level, notes, lock_id, lock_type):
     if lock_type is not None:
         parameters.type = lock_type
 
-def update_lock(name, resource_group_name=None,
-                level=None, notes=None, lock_id=None, lock_type=None):
+def update_lock(name, resource_group_name=None, level=None, notes=None):
+    '''
+    :param name: The name of the lock
+    :type name: str
+    :param resource_group_name: The name of the resource group, either the lock is on this group, or a
+    resource in this group.
```

Comment:
It sounds like resource_group_name should be required and not have a default value of None. This is another issue pointed out in #2694. It does not seem like you could identify any lock with just the name (unless they are globally unique).